libvast/vast/fbs/index.fbs

```diff
@@ -0,0 +1,33 @@
+include "meta_index.fbs";
+include "partition.fbs";
+include "uuid.fbs";
+include "version.fbs";
+
+namespace vast.fbs;
+
+table LayoutStatistics {
+	/// Name of the layout.
+	name: string;
+
+	/// Number of events of that type.
+	count: uint64;
+}
+
+/// The persistent state of the index.
+table Index {
+  /// The version of the index state.
+  version: Version;
+
+  /// Synopsis structures for the contained partitions.
+  meta_index: MetaIndex;
+
+  /// The contained partition UUIDs.
+  partitions: [UUID];
```

Comment:
We *could*, but it seems useful to have a sanity check of the state we expect to be present on-disk. Otherwise we would have no warning if partitions are present on disk but not in the meta index, or the other way round if partitions are suddenly missing after a restart.

Reply:
I see that, makes sense.